service/src/java/org/apache/hive/service/cli/OperationHandle.java

```diff
@@ -25,6 +25,7 @@
   private final OperationType opType;
   private final TProtocolVersion protocol;
   private boolean hasResultSet = false;
+  private boolean hasResultSetIsSet = false;
```

Comment:
Why do we need hasResultSetIsSet? isn't it enough to keep default as false for hasResultSet if it is null in OperationHandle?

Reply:
ok got it... btw, can we use Boolean object instead of boolean primitive type so that we can set it to null if not set. This will avoid extra variable hasResultSetIsSet.

Reply:
done